src/traversal.js

```diff
@@ -36,18 +39,21 @@ var enterNode = function(node) {
 var exitNode = function(node) {
   var data = getData(node);
   var lastVisitedChild = data.lastVisitedChild;
+  var lastChild = node.lastChild;
 
-  if (node.lastChild === lastVisitedChild) {
+  if (lastChild === lastVisitedChild) {
     return;
   }
 
-  while (node.lastChild !== lastVisitedChild) {
-    node.removeChild(node.lastChild);
-  }
+  while (lastChild !== lastVisitedChild) {
+    var key = getKey(lastChild);
+    if (key) {
+      registerChild(node, key, undefined);
```

Comment:
> We would want to delete the key off the mapping

But that'll guarantee the map goes into dictionary mode.

>  so that its size doesn't keep growing

For what I think would be the normal case (conditionally showing the same key'd element), this won't grow at all. For dynamic incrementing keys, bookkeeping could be easily added to prevent it.